### PR TITLE
Bl2501 ckeditor undo integration

### PR DESCRIPTION
**Pull request by hatton**
Fix BL-2501 Prefer ckeditor's undo if available. Else use document's instead of geckofx's.

This seems to fix the crash of BL-2501 and seems to give mostly the desired undo behavior even of stuff ckeditor does (e.g. formatting). However it falls should of good "canUndo()" detection, so if all you do is change formatting, the Undo button is not disabled. If however you do anything that does cause it to enable, it will undo formatting in the undo stack as you'd expect.

While we're at it, it's normally easy enough to ask the browser to undo from within javascript, so I changed it to do that rather than leave it to c# to do that.

### CODE REVIEW

File: src/BloomBrowserUI/bookEdit/js/calledByCSharp.ts [hunk 1]
```diff
@@ -2,20 +2,44 @@
...
+    }
+    /* I couldn't find a way to ask ckeditor if it is ready to do an undo.
+      The "canUndo()" is misleading; what it appears to mean is, can this command (undo) be undone?*/
+
+    /*  var ckEditorUndo = this.ckEditorUndoCommand();
+        if (ckEditorUndo === null) return 'fail';
+        return ckEditorUndo.canUndo() ? 'yes' : 'no';
+    */
+
+      return "fail"; //go ask the browser
```
Comment (StephenMcConnel) [inline src/BloomBrowserUI/bookEdit/js/calledByCSharp.ts]: So this particular method didn't really change functionally, but has some sample code that doesn't work left in to keep others from going down that rabbit trail.

**Comment by StephenMcConnel** [comment]
The new logic looks good to me.

File: src/BloomBrowserUI/bookEdit/js/calledByCSharp.ts (hunk 1 above)
Comment (hatton) [inline src/BloomBrowserUI/bookEdit/js/calledByCSharp.ts]: Yes. Do you think it'd be better without that?

Comment (StephenMcConnel) [inline src/BloomBrowserUI/bookEdit/js/calledByCSharp.ts]: I think it's fine as is with the commented code showing what doesn't work, in line with the preceding comment.  If we decide to work more in this area, it at least provides an example of what *doesn't* work.

**Pull request merged**